data_collection/gazette/spiders/es_cariacica.py

```diff
@@ -0,0 +1,53 @@
+import datetime
+import re
+from urllib.parse import urlencode
+
+from dateutil.rrule import WEEKLY, rrule
+from scrapy.http import Request
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class EsCariacicaSpider(BaseGazetteSpider):
+    name = "es_cariacica"
+    TERRITORY_ID = "3201308"
+    start_date = datetime.date(2014, 7, 1)
+    BASE_URL = "https://www.cariacica.es.gov.br/publicacoes/diario-oficial/"
+
+    def start_requests(self):
+        search_period = rrule(freq=WEEKLY, dtstart=self.start_date, until=self.end_date)
+
+        for date in search_period:
+            start = date.strftime("%Y-%m-%d")
+            last_day = date + datetime.timedelta(days=6)
+            if last_day.date() > self.end_date:
+                end = self.end_date.strftime("%Y-%m-%d")
+            else:
+                end = last_day.strftime("%Y-%m-%d")
+            interval = {"inicio": start, "fim": end}
+            url = f"{self.BASE_URL}?{urlencode(interval)}"
+            yield Request(url=url, callback=self.parse_week)
+
+    def parse_week(self, response):
+        date_cards = response.css("div.card div.card-header")
+
+        for card in date_cards[1:]:
+            date = card.xpath("./b/text()").get()
+            date = datetime.datetime.strptime(date, "%d/%m/%Y").date()
+
+            gazettes = card.xpath("./following-sibling::table/tr")
+            for gazette in gazettes[1:]:
+                link = gazette.xpath("./td[4]/a/@href").get()
```

Comment:
```suggestion
                link = response.urljoin(gazette.xpath("./td[4]/a/@href").get())
                if "https" not in link: link = "https" + link[4:]    
```

e mais abaixo no código, a linha 49 fica apenas: 
```
file_urls=[link],
```

isso é o jeito certo de lidar com a situação ou foi gambiarra?

Reply:
from urllib.parse import urlparse
u = urlparse('//www.cwi.nl:80/%7Eguido/Python.html')
u
ParseResult(scheme='', netloc='www.cwi.nl:80', path='/%7Eguido/Python.html',
            params='', query='', fragment='')
u._replace(scheme='http')
ParseResult(scheme='http', netloc='www.cwi.nl:80', path='/%7Eguido/Python.html',
            params='', query='', fragment='')
Giulio Carvalho15:44
url = "lalalalal"
parsed_url = urlparse(url)
fixed_url = urlunparse(parsed_url._replace(scheme="https"))
from urllib.parse import urlparse, urlunparse
Giulio Carvalho15:45
fixed_url = urlunparse(urlparse(url)._replace(scheme="https"))